### DISCUSSION TITLE
What is the most secure way to migrate from Rollout to Deployment without downtime?

### DISCUSSION THREAD

**Discussion by AviorSchreiber** [argoproj/argo-rollouts#2647] [Q&A]
Hi,
I'm trying to migrate Argo Rollout to Deployment my routing traffic is nginx-ingress.

I have created a new deployment side by side to the rollout as the [doc](https://argoproj.github.io/argo-rollouts/migrating/
) says.
but since that, the active service has been modified with an extra selector controlled by Argo Rollout (pod-template-hash),
The deployment pods are not joining the active service

I have also tried to create a different service side by side with the rollout active service for the deployment but then I failed with ingress configuration,
since there is already an ingress with the same configuration pointing to the rollout-active-service.

I have also tried to patch the service during the helm deployment using a custom Job with a pre-install hook, 
but then the Argo Rollout controller remodified them right away.

What do I miss here? 
  
**Version**
k8s: 1.24
ArgoCD: v2.5.2+148d8da
ArgoRollout: v1.4.0

**Comment by jsirianni** [comment]
I know this has been open for > 1 year but I have the same question. As long as your Rollout deploys correctly, you should not have downtime AFAIK.

It would be nice if we could allow the service to match the old Deployment and Rollout at the same time, with some option on the Rollout spec.strategy

**Comment by ptercero** [comment]
To me It would be also useful what @jsirianni mentions. A way to configure rollouts to un-patch de service BEFORE all rollout replicas are downscaled to 0. That way the svc can match both Deployment and Rollout PODs during the _Rollout->Deployment_ migration without downtime.
